parsec.cabal

```diff
@@ -86,7 +85,7 @@ library
         UndecidableInstances
 
     ghc-options: -Wall
-    if impl(ghc >= 8.0)
+    if impl(ghc >= 8.0) || impl(mhs)
```

Comment:
I actually dropped the support for pre GHC-8.0, so this conditional's else-branch can be removed. Do you mind?

I'm not sure whether Hackage is using new enough `Cabal` to accept `impl(mhs)` .

Reply:
You can remove the else branch if you want.

Cabal accepts mhs as a valid compiler now, I believe.

Reply:
Hmm. Maybe the CI is using some older cabal?

Reply:
> Cabal accepts mhs as a valid compiler now, I believe.

newer `Cabal` does. But `hackage-server` still uses old `Cabal`, so this change could not be released. (And it might take quite a long time).

Reply:
At the moment Hackage uses Cabal 3.12.1.0, which knows about MHS: 
* https://hackage.haskell.org/package/Cabal-syntax-3.12.1.0/docs/Distribution-Compiler.html#t:CompilerFlavor
* https://github.com/haskell/cabal/blob/master/release-notes/Cabal-3.12.1.0.md

Reply:
@Bodigrim good to know. But given a thought, if the thing can be written without `impl(mhs)`, it should.

You help me, I help you. I'm sorry, but I don't care about `MicroHS`, so if I can be unaware I will.

That said, the first moment the CI breaks, I will remove it; so adding CI for microhs is somewhat unnecessary too.

Reply:
If there is no support for pre-8.0 GHC then the condition can be simplified and it will work with MicroHs.

Reply:
I started this discussion with saying

> I actually dropped the support for pre GHC-8.0, so this conditional's else-branch can be removed. Do you mind?

---

This left a bad taste. I removed the conditional, so AFAICT this patch has only `MonoLocalBinds` changes which are relevant.